src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/ISymbolExtensions.cs

```diff
@@ -261,7 +261,7 @@ public static bool IsNormalAnonymousType([NotNullWhen(returnValue: true)] this I
             => symbol.IsAnonymousType() && !symbol.IsDelegateType();
 
         public static bool IsAnonymousDelegateType([NotNullWhen(returnValue: true)] this ISymbol? symbol)
-            => symbol.IsAnonymousType() && symbol.IsDelegateType();
+            => symbol.IsDelegateType() && (symbol.IsAnonymousType() || !symbol.CanBeReferencedByName);
```

Comment:
vb and C# are a bit weird here.   VB represents anonymous lambdas as actual .IsAnonymousType.  C# does not, but instead has CanBeReferencedByName=false.

Reply:
Was there an explicit decision here by the compiler team to use `CanBeReferencedByName` over `IsAnonymousType` as the api shape? Wish these could be consistent

Reply:
@cston any thoughts on why VB/C# differ here wrt anonymous delegates?

Reply:
It feels like `SynthesizedDelegateSymbol.IsAnonymousType` should return `true`. Please feel free to change that or log an issue.

Reply:
I'd agree with Chuck.

Reply:
[https://github.com/dotnet/roslyn/issues/55292](https://github.com/dotnet/roslyn/issues/55292)